qr_generator/lib/presentation/qr_scanner/scanner_screen.dart

```diff
@@ -0,0 +1,61 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:qr_code_scanner/qr_code_scanner.dart';
+import 'package:qr_generator/domain/qr_scanner/cubit/scanner_cubit.dart';
+
+class QRScannerScreen extends StatelessWidget {
+  const QRScannerScreen({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider<ScannerCubit>(
+      create: (_) => ScannerCubit(),
+      child: _QRScanner(),
+    );
+  }
+}
+
+class _QRScanner extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    final GlobalKey qrKey = GlobalKey(debugLabel: 'QR');
+    return Scaffold(
+      appBar: AppBar(
+        title: Text('QR Code'),
+      ),
+      body: BlocBuilder<ScannerCubit, ScannerState>(
+        builder: (_, state) => Column(
+          children: [
+            Expanded(
+              flex: 7,
+              child: QRView(
+                key: qrKey,
+                overlay: QrScannerOverlayShape(
+                  borderColor: Theme.of(context).accentColor,
+                  borderWidth: 8,
+                  borderRadius: 10,
+                ),
+                onQRViewCreated: (QRViewController controller) => {
+                  controller.scannedDataStream.listen((Barcode scanData) {
+                    context.read<ScannerCubit>().processBarcode(scanData);
```

Comment:
bug: this fires off tens of times a second and our ValidScannerState looks like this:
```date
class ValidScanState extends ScannerState {
  ValidScanState({
    required this.scanValue,
  });

  final Barcode scanValue;

  @override
  List<Object> get props => [scanValue];
}
```
The issue is that Equatable basically runs code that looks like this:
```
 @override
  bool operator ==(Object other) =>
    identical(this, other) ||
    other is ValidScanState &&
    scanValue == other.scanValue
```
except that `scanValue == other.scanValue` is not true since that object is not an immutable object. It would be this causes us to generate new classes and UI over and over. It would be better to unwrap the values we received from the `scanData` and pass them into our own object.